Accept all docker compose filenames on default (#138)

yarjor opened the pull request:
- Change `getComposeFiles` behavior to not throw an error if a compose file was not found (and only filter it out), and throw an error only if all give files were not found.
- Change the compose-file default to be all 4 docker compose filenames specified by [Docker's official documentation](https://docs.docker.com/compose/intro/compose-application-model/) . Because of the previous change, this will now cause the action to behave on default similarily to the `docker compose` command (in order to minimize confusion when using default parameters) - try to found files with all 4 possible names.
  -  One difference between this implementation and `docker compose`'s behavior that should be noted is that if multiple files exists (e.g `compose.yaml` and `docker-compose.yml`), the action will use both of them while `docker compose` will choose one file and has an order of preference.
- Updated tests accordingly

neilime commented:
Hi, thanks for your PR. 
IMO, this is good to match with docker compose specs.

But it is not good to had a breaking change for this (do not check if file exist anymore).

Maybe we can leave the input empty, if so we apply the defaults files and ignore exists check. If input is provided keep the original behavior

neilime reviewed (changes requested):
Hi, thanks for your PR. 
IMO, this is good to match with docker compose specs.

But it is not good to had a breaking change for this (do not check if file exist anymore).

Maybe we can leave the input empty, if so we apply the defaults files and ignore exists check. If input is provided keep the original behavior

yarjor commented:
> Hi, thanks for your PR. IMO, this is good to match with docker compose specs.
> 
> But it is not good to had a breaking change for this (do not check if file exist anymore).
> 
> Maybe we can leave the input empty, if so we apply the defaults files and ignore exists check. If input is provided keep the original behavior

Got you, I changed it back to throw an error if a file was not found, and added a separate flow to search for the default compose filenames if the input was empty. This time I also implementing choosing based on the order of preference specified in the official docs (compose > docker-compose, yaml > yml).

BTW, do note that the CI action has permission issues because this pull request is coming from a fork, so the actual action currently isn't tested - I made sure the TS tests pass of course. 😊

neilime commented on README.md:
The readme file is auto generated from action.yml, you don't have to edit it.
However, can you update compose-file input description in action.yml

neilime commented on action.yml:
```diff
@@ -12,7 +12,7 @@ inputs:
   compose-file:
     description: "Path to compose file(s). It can be a list of files. It can be absolute or relative to the current working directory (cwd)."
```
Can you update description to explain the new behavior

neilime commented on src/services/input.service.ts:
```diff
@@ -49,21 +49,38 @@ export class InputService {
+      // If no compose files are provided, check for all possible compose files by order of preference specified
+      // in official docker compose specs
+      composeFiles = ["compose.yaml", "compose.yml", "docker-compose.yaml", "docker-compose.yml"].filter((composeFile: string) => {
+        const possiblePaths = [join(cwd, composeFile), composeFile];
+
+        for (const path of possiblePaths) {
+          if (existsSync(path)) {
+            return true;
+          }
         }
-      }
 
-      throw new Error(`Compose file not found in "${possiblePaths.join('", "')}"`);
-    });
+        return false;
+      });
```
- Can you move it in a new dedicated private method like getExistingDefaultComposeFile.
- Return the first existing file instead of checking the whole list
- Throws an error if none found

neilime commented on .github/workflows/__check-action.yml:
```diff
@@ -29,6 +29,30 @@ jobs:
+
+      - name: "Assert: docker-compose.yaml is used"
+        run: |
+          docker compose -f ./test/docker-compose.yml ps
```
How is it possible it works as the file /test/docker-compose.yml has been renamed?
